src/main/java/at/hannibal2/skyhanni/features/garden/visitor/GardenVisitorSupercraft.kt

```diff
@@ -0,0 +1,119 @@
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.ItemUtils.itemName
+import at.hannibal2.skyhanni.utils.NEUInternalName
+import at.hannibal2.skyhanni.utils.NEUInternalName.Companion.asInternalName
+import at.hannibal2.skyhanni.utils.NEUItems
+import at.hannibal2.skyhanni.utils.NEUItems.getItemStack
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import io.github.moulberry.notenoughupdates.events.ReplaceItemEvent
+import io.github.moulberry.notenoughupdates.events.SlotClickEvent
+import io.github.moulberry.notenoughupdates.util.Utils
+import net.minecraft.entity.player.InventoryPlayer
+import net.minecraftforge.fml.common.eventhandler.EventPriority
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration.Companion.seconds
+
+class GardenVisitorSupercraft {
+
+    private val isSupercraftEnabled get() = VisitorAPI.config.shoppingList.showSuperCraft
+
+    private var hasIngredients = false;
+    private var lastClick = SimpleTimeMark.farPast()
+    private var lastSuperCraftMaterial = ""
+
+    private val superCraftItem by lazy {
+        val neuItem = "GOLD_PICKAXE".asInternalName().getItemStack()
+        Utils.createItemStack(
+            neuItem.item,
+            "§bSuper Craft",
+            "§7You have the items to craft",
+            "§7Click me to open the super crafter!"
+        )
+    }
+
+    @SubscribeEvent
+    fun onInventoryClose(event: InventoryCloseEvent) {
+        if (hasIngredients) {
+            hasIngredients = false
+        }
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    fun onVisitorOpen(event: VisitorOpenEvent) {
+        val visitor = event.visitor
+        val offerItem = visitor.offer!!.offerItem
+
+        val lore = offerItem.getLore()
+        for (line in lore) {
+            if (line == "§7Items Required:") continue
+            if (line.isEmpty()) break
+
+            val pair = ItemUtils.readItemAmount(line)
+            if (pair == null) {
+                ErrorManager.logErrorStateWithData(
+                    "Could not read Shopping List in Visitor Inventory", "ItemUtils.readItemAmount returns null",
+                    "line" to line,
+                    "offerItem" to offerItem,
+                    "lore" to lore,
+                    "visitor" to visitor
+                )
+                continue
+            }
+            val (itemName, amount) = pair
+            val internalName = NEUInternalName.fromItemName(itemName)
+//
+//            GET SUPERCRAFT FOR SACKS CODE
+//
+            if (isSupercraftEnabled) {
+                val item =
+                    NEUItems.getRecipes(internalName).first { !it.ingredients.first().internalItemId.contains("PEST") }
+                val ingredients = item.ingredients
+                val ingredientReqs = mutableMapOf<String, Int>()
+                for (ingredient in ingredients) {
+                    val ing = ingredient.serialize().split(":")
```

Comment:
instead of using serialize and split, you can work with count and internalItemId directly